daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/types/Ledger.scala

```diff
@@ -474,7 +474,7 @@ object Ledger {
   def initialLedger(t0: Time.Timestamp): Ledger =
     Ledger(
       currentTime = t0,
-      scenarioStepId = ScenarioTransactionId(0),
+      scenarioStepId = ScenarioTransactionId(-1), //we're starting with -1 so the first transaction gets 0! This is important to avoid clashes on the first real transaction coming through the Ledger API.
```

Comment:
@remyhaemmerle-da @dajmaki @gerolf-da where do you think it would the be to protect it with a unit test?

Reply:
There's `ScenarioRunnerTest`. Maybe add it there?

Reply:
Wouldn't it be better to have a test on the `Ledger` itself? I create a new test for it.